core/trino-spi/src/main/java/io/trino/spi/block/AbstractVariableWidthBlock.java

```diff
@@ -120,6 +120,23 @@ public void writeBytesTo(int position, int offset, int length, BlockBuilder bloc
         blockBuilder.writeBytes(getRawSlice(position), getPositionOffset(position) + offset, length);
     }
 
+    public void writeBytesTo(int[] positions, int positionsCount, BatchByteWriter byteWriter)
+    {
+        int[] lengths = new int[positionsCount];
+        int[] offsets = new int[positionsCount];
+        for (int i = 0; i < positionsCount; i++) {
+            int position = positions[i];
+            if (!isNull(position)) {
```

Comment:
There is no need for `isNull` check. Slice length will be 0 and position offset will be same as the last position's offset.